orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -305,12 +334,17 @@ def _add_dynamic_shape(name, input):
         return dynamic_axes
 
     def _add_input(name, input, onnx_graph, onnx_graph_input_names):
-        if input is not None and (onnx_graph is None or name in onnx_graph_input_names):
-            if input.requires_grad:
-                input_names_require_grad.append(name)
+        if input is not None:
+            if (onnx_graph is None or name in onnx_graph_input_names):
+                if isinstance(input, torch.Tensor):
+                    if input.requires_grad:
```

Comment:
4 levels of `if` give a hard time for the optimizer to properly parallelize or speed up the CPU pipeline. If you can optimize this somehow, that would be nice

Reply:
Unfortunately, I am not able to optimize this. However, I may be able to exit early and reduce the visibility of 4 checks.